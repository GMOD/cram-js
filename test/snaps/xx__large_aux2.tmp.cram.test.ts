import { testFile } from '../testUtil'
testFile('xx#large_aux2.tmp.cram')
