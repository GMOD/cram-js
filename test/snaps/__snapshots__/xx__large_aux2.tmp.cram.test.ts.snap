// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole xx#large_aux2.tmp.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 44,
      "_size": 18,
      "alignmentSpan": 0,
      "crc32": 3519680774,
      "landmarks": [
        0,
        119,
      ],
      "length": 187,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 44,
        "_size": 18,
        "alignmentSpan": 0,
        "crc32": 3519680774,
        "landmarks": [
          0,
          119,
        ],
        "length": 187,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 32,
          },
          "parametersBytes": 1,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {
        "aaC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 6381891,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
        "abC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 6382147,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
        "acC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 6382403,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
        "adC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 6382659,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
        "aeC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 6382915,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
        "afC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 6383171,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
        "agC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 6383427,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
        "ahC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 6383683,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
        "aiC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 6383939,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
        "ajC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 6384195,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
      },
      "tagIdsDictionary": [
        [
          "aaC",
        ],
        [
          "aaC",
          "abC",
        ],
        [
          "aaC",
          "abC",
          "acC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
          "afC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
          "afC",
          "agC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
          "afC",
          "agC",
          "ahC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
          "afC",
          "agC",
          "ahC",
          "aiC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
          "afC",
          "agC",
          "ahC",
          "aiC",
          "ajC",
        ],
      ],
    },
    "containerHeader": {
      "_endPosition": 251,
      "_size": 20,
      "alignmentSpan": 1,
      "crc32": 1267951824,
      "landmarks": [
        534,
        970,
      ],
      "length": 1508,
      "numBases": 10,
      "numBlocks": 28,
      "numLandmarks": 2,
      "numRecords": 10,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 251,
        "_size": 20,
        "alignmentSpan": 1,
        "crc32": 1267951824,
        "landmarks": [
          534,
          970,
        ],
        "length": 1508,
        "numBases": 10,
        "numBlocks": 28,
        "numLandmarks": 2,
        "numRecords": 10,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a1",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
            },
            "templateSize": 0,
            "uniqueId": 791,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a2",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
            },
            "templateSize": 0,
            "uniqueId": 792,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a3",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
            },
            "templateSize": 0,
            "uniqueId": 793,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a4",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
            },
            "templateSize": 0,
            "uniqueId": 794,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a5",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
            },
            "templateSize": 0,
            "uniqueId": 795,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a6",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
              "af": 1,
            },
            "templateSize": 0,
            "uniqueId": 796,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a7",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
              "af": 1,
              "ag": 1,
            },
            "templateSize": 0,
            "uniqueId": 797,
          },
        ],
        "header": {
          "_endPosition": 853,
          "_size": 63,
          "compressedSize": 59,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            1,
            7,
            0,
            11,
            10,
            11,
            12,
            32,
            224,
            97,
            101,
            67,
            224,
            97,
            100,
            67,
            224,
            97,
            98,
            67,
            224,
            97,
            97,
            67,
            224,
            97,
            103,
            67,
            224,
            97,
            99,
            67,
            224,
            97,
            102,
            67,
            255,
            255,
            255,
            255,
            15,
            127,
            197,
            98,
            112,
            231,
            167,
            15,
            168,
            26,
            89,
            53,
            183,
            46,
            172,
            190,
            41,
          ],
          "contentId": 0,
          "contentPosition": 790,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 587337047,
          "parsedContent": {
            "_endPosition": 310,
            "_size": 59,
            "contentIds": [
              11,
              12,
              32,
              6382915,
              6382659,
              6382147,
              6381891,
              6383427,
              6382403,
              6383171,
            ],
            "md5": [
              127,
              197,
              98,
              112,
              231,
              167,
              15,
              168,
              26,
              89,
              53,
              183,
              46,
              172,
              190,
              41,
            ],
            "numBlocks": 11,
            "numContentIds": 10,
            "numRecords": 7,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 1,
            "refSeqStart": 1,
          },
          "uncompressedSize": 59,
        },
      },
      {
        "block": {
          "_endPosition": 862,
          "_size": 4,
          "compressedSize": 0,
          "compressionMethod": "raw",
          "content": Uint8Array [],
          "contentId": 0,
          "contentPosition": 858,
          "contentType": "CORE_DATA",
          "crc32": 4059825967,
          "uncompressedSize": 0,
        },
      },
      {
        "block": {
          "_endPosition": 912,
          "_size": 45,
          "compressedSize": 41,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            97,
            49,
            0,
            97,
            50,
            0,
            97,
            51,
            0,
            97,
            52,
            0,
            97,
            53,
            0,
            97,
            54,
            0,
            97,
            55,
            0,
          ],
          "contentId": 11,
          "contentPosition": 867,
          "contentType": "EXTERNAL_DATA",
          "crc32": 1885374484,
          "uncompressedSize": 21,
        },
      },
      {
        "block": {
          "_endPosition": 943,
          "_size": 26,
          "compressedSize": 22,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            2,
            2,
            2,
            2,
            2,
            2,
            2,
          ],
          "contentId": 12,
          "contentPosition": 917,
          "contentType": "EXTERNAL_DATA",
          "crc32": 1278520618,
          "uncompressedSize": 7,
        },
      },
      {
        "block": {
          "_endPosition": 979,
          "_size": 31,
          "compressedSize": 27,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            0,
            1,
            2,
            3,
            4,
            5,
            6,
          ],
          "contentId": 32,
          "contentPosition": 948,
          "contentType": "EXTERNAL_DATA",
          "crc32": 3938949832,
          "uncompressedSize": 7,
        },
      },
      {
        "block": {
          "_endPosition": 1014,
          "_size": 27,
          "compressedSize": 23,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            1,
            1,
            1,
          ],
          "contentId": 6382915,
          "contentPosition": 987,
          "contentType": "EXTERNAL_DATA",
          "crc32": 825089048,
          "uncompressedSize": 3,
        },
      },
      {
        "block": {
          "_endPosition": 1048,
          "_size": 26,
          "compressedSize": 22,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            1,
            1,
            1,
            1,
          ],
          "contentId": 6382659,
          "contentPosition": 1022,
          "contentType": "EXTERNAL_DATA",
          "crc32": 4140596132,
          "uncompressedSize": 4,
        },
      },
      {
        "block": {
          "_endPosition": 1086,
          "_size": 30,
          "compressedSize": 26,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            1,
            1,
            1,
            1,
            1,
            1,
          ],
          "contentId": 6382147,
          "contentPosition": 1056,
          "contentType": "EXTERNAL_DATA",
          "crc32": 3522974118,
          "uncompressedSize": 6,
        },
      },
      {
        "block": {
          "_endPosition": 1120,
          "_size": 26,
          "compressedSize": 22,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            1,
            1,
            1,
            1,
            1,
            1,
            1,
          ],
          "contentId": 6381891,
          "contentPosition": 1094,
          "contentType": "EXTERNAL_DATA",
          "crc32": 2036669967,
          "uncompressedSize": 7,
        },
      },
      {
        "block": {
          "_endPosition": 1153,
          "_size": 25,
          "compressedSize": 21,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            1,
          ],
          "contentId": 6383427,
          "contentPosition": 1128,
          "contentType": "EXTERNAL_DATA",
          "crc32": 1959985495,
          "uncompressedSize": 1,
        },
      },
      {
        "block": {
          "_endPosition": 1187,
          "_size": 26,
          "compressedSize": 22,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            1,
            1,
            1,
            1,
            1,
          ],
          "contentId": 6382403,
          "contentPosition": 1161,
          "contentType": "EXTERNAL_DATA",
          "crc32": 1338011480,
          "uncompressedSize": 5,
        },
      },
      {
        "block": {
          "_endPosition": 1221,
          "_size": 26,
          "compressedSize": 22,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            1,
            1,
          ],
          "contentId": 6383171,
          "contentPosition": 1195,
          "contentType": "EXTERNAL_DATA",
          "crc32": 3029976740,
          "uncompressedSize": 2,
        },
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a8",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
              "af": 1,
              "ag": 1,
              "ah": 1,
            },
            "templateSize": 0,
            "uniqueId": 1234,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a9",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
              "af": 1,
              "ag": 1,
              "ah": 1,
              "ai": 1,
            },
            "templateSize": 0,
            "uniqueId": 1235,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "aA",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
              "af": 1,
              "ag": 1,
              "ah": 1,
              "ai": 1,
              "aj": 1,
            },
            "templateSize": 0,
            "uniqueId": 1236,
          },
        ],
        "header": {
          "_endPosition": 1301,
          "_size": 75,
          "compressedSize": 71,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            1,
            3,
            7,
            14,
            13,
            11,
            12,
            32,
            224,
            97,
            106,
            67,
            224,
            97,
            101,
            67,
            224,
            97,
            100,
            67,
            224,
            97,
            98,
            67,
            224,
            97,
            97,
            67,
            224,
            97,
            105,
            67,
            224,
            97,
            103,
            67,
            224,
            97,
            99,
            67,
            224,
            97,
            102,
            67,
            224,
            97,
            104,
            67,
            255,
            255,
            255,
            255,
            15,
            127,
            197,
            98,
            112,
            231,
            167,
            15,
            168,
            26,
            89,
            53,
            183,
            46,
            172,
            190,
            41,
          ],
          "contentId": 0,
          "contentPosition": 1226,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 1817932596,
          "parsedContent": {
            "_endPosition": 322,
            "_size": 71,
            "contentIds": [
              11,
              12,
              32,
              6384195,
              6382915,
              6382659,
              6382147,
              6381891,
              6383939,
              6383427,
              6382403,
              6383171,
              6383683,
            ],
            "md5": [
              127,
              197,
              98,
              112,
              231,
              167,
              15,
              168,
              26,
              89,
              53,
              183,
              46,
              172,
              190,
              41,
            ],
            "numBlocks": 14,
            "numContentIds": 13,
            "numRecords": 3,
            "recordCounter": 7,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 1,
            "refSeqStart": 1,
          },
          "uncompressedSize": 71,
        },
      },
    ],
  },
]
`;
